File: kotlin/hamming/src/main/kotlin/Hamming.kt
import java.lang.IllegalArgumentException

class Hamming {
    companion object {
        fun compute(s1: String, s2: String): Int {
            require(s1.length == s2.length) { "left and right strands must be of equal length." }

            return s1.zip(s2).filter { pair -> pair.first != pair.second }.size
        }
    }
}

File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    local distance = 0

    if #a ~= #b then
        error("strands must be of equal length")
    end
    
    for i = 1, #a do 
        local char_a = a:sub(i, i)
        local char_b = b:sub(i, i)

        if (char_a ~= char_b) then
            distance = distance + 1
        end
    end

    return distance
end

return Hamming
